ansible/roles/jenkins-worker/templates/ibmi_start.j2

```diff
@@ -1,15 +1,18 @@
-#!/bin/sh
-
-su -s - iojs <<'EOF'
+#!/QOpenSys/pkgs/bin/bash
+SBMJOB_OPTS="JOB(JENKINS_CI) USER(IOJS)" 
 
 export PATH=/QOpenSys/pkgs/lib/ccache:/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
-export HOME={{ home }}/{{ server_user }}
+export JENKS_HOME={{ home }}/{{ server_user }}
 export NODE_COMMON_PIPE="$HOME/test.pipe"
 export NODE_TEST_DIR="$HOME/tmp"
 export JOBS=4
 export OSTYPE=ibmi
+export QIBM_USE_DESCRIPTOR_STDIO=N
+
+unset LC_ALL
+
+export START_CMD="{{ java_path[os] }} -Xmx{{ server_ram|default('128m') }} -jar {{ home }}/{{ server_user }}/slave.jar -secret {{ secret }} -jnlpUrl {{ jenkins_url }}/computer/{{ inventory_hostname }}/slave-agent.jnlp >{{ home }}/{{ server_user }}/jenkins.log 2>&1 "
+
+echo $START_CMD
 
-{{ java_path[os] }} -Xmx{{ server_ram|default('128m') }} \
-    -jar {{ home }}/{{ server_user }}/slave.jar -secret {{ secret }} \
-    -jnlpUrl {{ jenkins_url }}/computer/{{ inventory_hostname }}/slave-agent.jnlp >{{ home }}/{{ server_user }}/jenkins.log 2>&1  &
-EOF
+exec /QOpenSys/usr/bin/system -kpiveO "SBMJOB CMD(CALL PGM(QP2SHELL2) PARM('/QOpenSys/usr/bin/sh' '-c' 'echo starting Jenkins slave && cd $JENKS_HOME && env && exec $START_CMD')) CPYENVVAR(*YES) PRTDEV(*USRPRF) ALWMLTTHD(*YES) $SBMJOB_OPTS"
```

Comment:
If we're using SBMJOB then none of the environment variables we set above will be inherited by the spawned job.

I also don't think we should be using QP2SHELL2, since it doesn't set up stdin, stdout, or stderr. (maybe it doesn't matter since we're redirecting anyway)

Reply:
It should get those environment variables because of the `CPYENVVAR(*YES)`, though PATH might be a special case when we hop back into PASE. 
I assumed the redirect eliminates the need for descriptor setup and verified with some simplified test cases